[Style] 모립세트 페이지 퍼블리싱

Ivoryeee opened the pull request:
<!-- PR 제목은 관련 이슈번호의 제목과 동일한 제목!! -->

## 🔥 Related Issues

- close #220 

## ✅ 작업 리스트

- [x] 모립세트 페이지 퍼블리싱

## 🔧 작업 내용

###  MoribSetPage.tsx 
default로 `MoribSet` 작성란을 두기 위해서 `MoribSetPage`에서 `moribSets`의 초기값을 다음과 같이 추가해 두었습니다. 
```typescript
	const [moribSets, setMoribSets] = useState<MoribSet[]>([
		{
			id: Date.now(),
			moribSetName: '',
			selectedColor: '#868C93',
			urlList: [],
		},
	]);
```


###  CategoryMoribSet.tsx - 모립세트 페이지의 좌측 카테고리 
```typescript
const CategoryMoribSet = ({
	moribSets,
	activeMoribSetId,
	setActiveMoribSetId,
	addMoribSet,
	deleteMoribSet,
}: CategoryMoribSetProps) => {
	const maxIconsToShow = 5;

	return (
		<div className="ml-[4.2rem] h-[99.6rem] w-[40.2rem] flex-shrink-0 rounded-[1.6rem] bg-gray-bg-03">
			<div className="mb-[2.4rem] mt-[2.2rem] flex items-center justify-between px-[2.8rem]">
				<h1 className="head-bold-24 text-white">내 모립세트</h1>
				<IconPlus className="cursor-pointer" onClick={addMoribSet} />
			</div>

			<div className="mx-[1.8rem] gap-[10px]">
				{moribSets.map((moribSet) => {
					const additionalIconsCount =
						moribSet.urlList.length > maxIconsToShow ? moribSet.urlList.length - maxIconsToShow : 0;

					return (
						<div
							key={moribSet.id}
							className={`mb-[8px] h-[8rem] w-[36.6rem] flex-col items-start justify-end rounded-[8px] bg-gray-bg-01 p-[1.4rem] ${activeMoribSetId === moribSet.id ? 'border-[1px] border-mint-01' : ''}`}
							onClick={() => setActiveMoribSetId(moribSet.id)}
						>
							<div className="flex content-between items-center self-stretch">
								<div className="flex w-[30.8rem] items-center gap-[6px]">
									<div
										className="h-[1.4rem] w-[1.4rem] rounded-[70px]"
										style={{ backgroundColor: moribSet.selectedColor }}
									/>
									<h2
										className={`body-semibold-16 w-[28.8rem] truncate ${moribSet.moribSetName ? 'text-white' : 'text-gray-03'}`}
									>
										{moribSet.moribSetName || '모립세트 이름을 입력해주세요.'}
									</h2>
								</div>

								<Dropdown.Root>
									<Dropdown.Trigger>
										<IconMeatBall className="cursor-pointer" />
									</Dropdown.Trigger>
									<Dropdown.Content
										boxShadow="shadow-none"
										className="top-[26px] w-[12.4rem] overflow-x-hidden overflow-y-hidden"
									>
										<Dropdown.Item label="할 일 삭제" textColor="red" onClick={() => deleteMoribSet(moribSet.id)} />
									</Dropdown.Content>
								</Dropdown.Root>
							</div>

							<div className="mt-[0.4rem] flex items-center gap-[6px]">
								{moribSet.urlList.slice(0, maxIconsToShow).map(({ faviconUrl }, url) => (
									<img key={url} src={faviconUrl} alt="`favicon" className="h-[2rem] w-[2rem] rounded-full" />
								))}
								{additionalIconsCount > 0 && (
									<div className="body-detail-reg-12 flex h-[2rem] w-[2rem] items-center justify-center rounded-[57px] bg-date-active text-white">
										+{additionalIconsCount}
									</div>
								)}
							</div>
						</div>
					);
				})}
			</div>
		</div>
	);
};
```
###  BoxMakeMoribSet.tsx - 모립세트 페이지의 우측 모립세트 추가창

#### InputMoribSetUrl - url 유효성 검사 
```typescript
export const urlRegex = /^(https?:\/\/)?([\w\-]+\.)+[a-zA-Z]{2,}(\/[\w#!:.?+=&%@!\-\/]*)?$/;

		if (!urlRegex.test(url)) {
			setAlert({ error: '알맞은 형식의 주소를 입력해주세요.' });
			return;
		}
```
`http`, `https` 또는 `www`로 시작하는 주소를 허용합니다

#### BoxMakeMoribSet - urlList로 처리 방법

```typescript
			const domain = new URL(url).hostname;
			const faviconUrl = `https://www.google.com/s2/favicons?domain=${domain}`;

			updateMoribSet('urlList', [
				...urlList,
				{
					siteName: domain.replace(/^www\./, '').replace(/\.[a-z]{2,}$/, ''),
					page: domain, //api로 받은 값
					url,
					faviconUrl,
				},
			]);
```
`URL` 객체를 사용해 `hostname`으로 페이지 이름을 가져오고, `siteName`은 `hostname`의 `www` 또는 `.com`과 같은 값을 필터링한 값을 보이도록 설정했습니다. `domain`은 탭 이름 get 하는 api로 추후 수정될 예정입니다


```typescript
const BoxMakeMoribSet = ({ moribSet, updateMoribSet }: BoxMakeMoribSetProps) => {
	const { moribSetName, selectedColor, urlList } = moribSet;
	const [isPaletteOpen, setIsPaletteOpen] = useState(false);
	const [selectedTabId, setSelectedTabId] = useState(CATEGORY_MODALTABS[0].id);

	const handleColorPalleteOpen = () => {
		setIsPaletteOpen((prev) => !prev);
	};

	const handleColorSelect = (color: string) => {
		updateMoribSet('selectedColor', color);
		setIsPaletteOpen(false);
	};

	const handleMoribNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		updateMoribSet('moribSetName', e.target.value);
	};

	const handleTabChange = (tab: number) => {
		setSelectedTabId(tab);
	};
	const handleDeleteUrlInfo = (urlToDelete: string) => {
		const updatedUrlList = urlList.filter((urlInfo) => urlInfo.url !== urlToDelete);
		updateMoribSet('urlList', updatedUrlList);
	};

	const addUrl = (url: string) => {
		try {
			const domain = new URL(url).hostname;
			const faviconUrl = `https://www.google.com/s2/favicons?domain=${domain}`;

			updateMoribSet('urlList', [
				...urlList,
				{
					siteName: domain.replace(/^www\./, '').replace(/\.[a-z]{2,}$/, ''),
					page: domain, //api로 받은 값
					url,
					faviconUrl,
				},
			]);
		} catch (error) {
			console.error('Invalid URL:', error);
		}
	};

	return (
		<div className="relative flex w-[132rem] flex-col items-start gap-[20px]">
			<div className="flex items-center gap-[14px]">
				<div className="flex items-center gap-[6px]">
					<div className="h-[3rem] w-[3rem] rounded-[31px]" style={{ backgroundColor: selectedColor }} />
					<ArrowSVGBtn direction={Direction.DOWN} onClick={handleColorPalleteOpen} />
				</div>
				<input
					value={moribSetName}
					onChange={handleMoribNameChange}
					placeholder="모립세트 이름을 입력해주세요."
					className="title-bold-36 placeholder-text-gray-03 w-[114rem] flex-shrink-0 bg-transparent text-white focus:outline-none"
				/>
			</div>

			<ColorPallete isOpen={isPaletteOpen} onSelectColor={handleColorSelect} />

			<div className="flex items-center gap-[3px] self-stretch">
				<CategoryTabSelect tabs={CATEGORY_MODALTABS} handleTabChange={handleTabChange} selectedTabId={selectedTabId} />
			</div>

			{selectedTabId === CATEGORY_MODALTABS[0].id && (
				<>
					<div className="relative w-full">
						<InputMoribSetUrl addUrl={addUrl} />
					</div>
					<TableMoribSet urlList={urlList}>
						<BoxUrlList urlList={urlList} onDelete={handleDeleteUrlInfo} />
					</TableMoribSet>
				</>
			)}
		</div>
	);
};

```

###  BoxRecommendService.tsx - 모립세트 페이지의 우측 하단 추천 서비스
```typescript
const BoxRecommendService = ({ addUrlToMoribSet }: BoxRecommendServiceProps) => {
	const [availableServices, setAvailableServices] = useState(recommendServices);

	const handleServiceClick = (service: { serviceName: string; url: string }) => {
		addUrlToMoribSet({
			siteName: service.serviceName.replace(/^www\./, '').replace(/\.[a-z]{2,}$/, ''),
			page: service.serviceName,
			url: service.url,
			faviconUrl: `https://www.google.com/s2/favicons?domain=${service.url}`,
		});
		setAvailableServices((prevServices) => prevServices.filter((item) => item.url !== service.url));
	};

	const carouselContainerRef = useRef<HTMLDivElement | null>(null);

	const scrollCarousel = (direction: 'left' | 'right') => {
		if (carouselContainerRef.current) {
			const scrollAmount = 1264;
			if (direction === 'left') {
				carouselContainerRef.current.scrollLeft -= scrollAmount;
			} else {
				carouselContainerRef.current.scrollLeft += scrollAmount;
			}
		}
	};

	return (
		<div className="relative h-[18.8rem] w-[132rem] flex-shrink-0 rounded-[16px] bg-gray-bg-03">
			<div className="mx-[2.8rem] flex w-[126.4rem] items-end justify-between self-stretch pt-[2.2rem]">
				<h2 className="head-bold-24 text-white">추천 서비스</h2>
				<div className="flex items-center">
					<ArrowSVGBtn
						className="flex h-[3.4rem] w-[3.4rem] items-center justify-center p-[0.5rem]"
						direction={Direction.LEFT}
						onClick={() => scrollCarousel('left')}
					/>
					<ArrowSVGBtn
						className="z-10 flex h-[3.4rem] w-[3.4rem] items-center justify-center p-[0.5rem]"
						direction={Direction.RIGHT}
						onClick={() => scrollCarousel('right')}
					/>
				</div>
			</div>

			<div
				ref={carouselContainerRef}
				className="mx-[2.8rem] my-[2.4rem] flex items-center gap-[14px] overflow-x-hidden"
			>
				{availableServices.map((service, index) => (
					<div
						key={index}
						className="flex w-[23.9rem] flex-shrink-0 items-center gap-[15px] rounded-[8px] bg-gray-bg-01 p-[2rem]"
						onClick={() => handleServiceClick(service)}
					>
						<img
							src={`https://www.google.com/s2/favicons?domain=${service.url}`}
							alt="favicon"
							className="h-[4.2rem] w-[4.2rem] rounded-full"
						/>
						<p className="subhead-bold-20 max-h-[8.4rem] overflow-hidden text-white">{service.serviceName}</p>
					</div>
				))}
				{/* 				<div
					className="absolute right-0 top-0 z-0 h-[18.8rem] w-[7.2rem] flex-shrink-0"
					style={{
						background: 'linear-gradient(90deg, rgba(51, 55, 63, 0.00) 0%, #33373F 100%)',
					}}
				/> */}
			</div>
		</div>
	);
};
``` 


## 🧐 새로 알게된 점
- index 대신 `id: Date.now()` 로 id를 구성해 id의 고유성을 보장할 수 있다 

## 🤔 궁금한 점
모립세트 리스트 삭제 후 현재 id의 바로 전 id를 참조하기 위해서 `activeMoribSetId` 상태를 추가해 두었습니다. 
```typescript
 const handleDeleteMoribSet = (id: number) => {
    const updatedMoribSets = moribSets.filter((moribSet) => moribSet.id !== id);
    setMoribSets(updatedMoribSets);

    if (activeMoribSetId === id) {
      if (updatedMoribSets.length > 0) {
        const newActiveMoribSetId = updatedMoribSets[updatedMoribSets.length - 1].id;
        setActiveMoribSetId(newActiveMoribSetId);
      } else {
        setActiveMoribSetId(null);
      }
    }
  };
```
원래 이런 식으로 로직을 구성해 업데이트된 모립세트의 가장 마지막 id를 가져와 Active 되게 하려 했는데 이렇게는 작동이 되지 않아 불가피하게 `useEffect`를 사용해 아래와 같이 수정했습니다. 

```typescript
	const handleDeleteMoribSet = (id: number) => {
		const updatedMoribSets = moribSets.filter((moribSet) => moribSet.id !== id);
		if (updatedMoribSets.length === 0) {
			return;
		}
		setMoribSets(updatedMoribSets);
	};

	useEffect(() => {
		if (moribSets.length === 0) {
			setActiveMoribSetId(null);
		} else if (!moribSets.find((set) => set.id === activeMoribSetId)) {
			setActiveMoribSetId(moribSets[moribSets.length - 1].id);
		}
	}, [moribSets, activeMoribSetId]);
```
이것보다 더 좋은 방법이 있을지 고민이 됩니다

## 📸 스크린샷 / GIF / Link

![moribSetPr1](https://github.com/user-attachments/assets/3d88af28-1cdd-457a-bc58-b6bd138deabd)
![moribSetPr2](https://github.com/user-attachments/assets/597d5fe5-ad77-4fa3-9adf-ba380c0dfac9)

Ivoryeee committed:
[style] 사이드바 모립세트 뷰 추가 (#220)

Ivoryeee committed:
[style] 모립세트 경로 추가 (#220)

Ivoryeee committed:
[style] tab 스타일 변경사항 반영 (#220)

Ivoryeee committed:
[style] 모립세트 페이지 퍼블리싱 (#220)

suwonthugger commented on src/pages/MoribSetPage/components/Box/BoxEachUrlInfo.tsx:
```diff
@@ -0,0 +1,28 @@
+				<p className="truncate">{urlInfo.siteName}</p>
+			</div>
+			<div className="body-reg-16 w-[31rem] text-left text-gray-04">
+				<p className="truncate">{urlInfo.page}</p>
```
p5: `overflow: hidden; text-overflow: ellipsis; white-space: nowrap;` 이거 다 따로 적용하고 있었는데, truncate가 있었군요... 와,, 저도 알려주시지

suwonthugger commented on src/pages/MoribSetPage/MoribSetPage.tsx:
```diff
@@ -0,0 +1,139 @@
+	const [activeMoribSetId, setActiveMoribSetId] = useState<number | null>(moribSets[0].id);
+
+	const handleAddMoribSet = () => {
+		const newSet: MoribSet = {
+			id: Date.now(),
+			moribSetName: '',
+			selectedColor: '#868C93',
+			urlList: [],
+		};
```
p5:  `Date.now()`로 유니크한 id 값을 생성해주셨군요 굿. 추후에 서버에서 데이터를 받아서 연동할 때는 `Date.now()` 대신  id를 서버에서 받은 데이터로 변경 해주면 좋을것 같아요.
서버에서 실제 데이터를 받아올 때도 `Date.now()`로 id값을 지정해주게 되면 모든 객체가 다 유니크해지기 때문에, 서버에서 실수로 중복 데이터를 보내도 잘못됐다는걸 바로 캐치할 수 없기 때문이에요. 요점 참고 부탁드립니다 굿굿~

Ivoryeee replied:
네 확인했습니다!

suwonthugger commented on src/pages/MoribSetPage/MoribSetPage.tsx:
```diff
@@ -0,0 +1,139 @@
+		);
+	};
+
+	const handleDeleteMoribSet = (id: number) => {
+		const updatedMoribSets = moribSets.filter((moribSet) => moribSet.id !== id);
+		if (updatedMoribSets.length === 0) {
+			return;
+		}
+		setMoribSets(updatedMoribSets);
+	};
```
p5: 엇 filter를 적용하면서 모든 모립세트가 삭제가 될수도 있을것 같은데 `updatedMoribSets.length === 0` 일 때는 왜 return을 하는지 궁금해요!

Ivoryeee replied:
최소 한 개 이상의 허용 서비스가 항시 남아있을 수 있도록 다음과 같은 코드를 수정하였습니다!

suwonthugger commented on src/pages/MoribSetPage/MoribSetPage.tsx:
```diff
@@ -0,0 +1,139 @@
+		);
+	};
+
+	useEffect(() => {
+		if (moribSets.length === 0) {
+			setActiveMoribSetId(null);
+		} else if (!moribSets.find((set) => set.id === activeMoribSetId)) {
+			setActiveMoribSetId(moribSets[moribSets.length - 1].id);
+		}
+	}, [moribSets, activeMoribSetId]);
```
p1: 이 부분은 `useEffect`를 사용하지않고, 모립세트를 삭제하는 경우에 상아님이 작성하신 아래 로직을 적용하는걸 제안드리고 싶은데 어떠신가요?

```javascript
if (moribSets.length === 0) {
			setActiveMoribSetId(null);
		} else if (!moribSets.find((set) => set.id === activeMoribSetId)) {
			setActiveMoribSetId(moribSets[moribSets.length - 1].id);
		}
```

KIMGEONHWI replied:
> p1: 이 부분은 `useEffect`를 사용하지않고, 모립세트를 삭제하는 경우에 상아님이 작성하신 아래 로직을 적용하는걸 제안드리고 싶은데 어떠신가요?
> 
> ```js
> if (moribSets.length === 0) {
> 			setActiveMoribSetId(null);
> 		} else if (!moribSets.find((set) => set.id === activeMoribSetId)) {
> 			setActiveMoribSetId(moribSets[moribSets.length - 1].id);
> 		}
> ```

p1: 저도 동의합니다. 현재 모립세트 삭제시 이전 내용이 화면에 깜빡이면서 보이는 문제가 발생하고 있습니다.

seueooo replied:
저도 동의합니다!

Ivoryeee replied:
pr에 작성해둔 바와 같이 모립세트 삭제 시 위의 코드를 추가하 경우 activeMoribSetId가 자동으로 업데이트되지 않는 문제가 있습니다. 현재 상태에서 useEffect를 사용하는 게 최선이라 생각이 들어 이와 같이 작성해 두었습니다.

suwonthugger commented on src/pages/MoribSetPage/MoribSetPage.tsx:
```diff
@@ -0,0 +1,139 @@
+				/>
+
+				<div className="ml-[4.2rem] mt-[6.8rem]">
+					{activeMoribSetId && activeMoribSet && (
+						<BoxMakeMoribSet
+							moribSet={activeMoribSet}
+							updateMoribSet={(key, value) => handleUpdateMoribSet(activeMoribSetId, key, value)}
+						/>
+					)}
```
p3: 혹시 activeMoribSet 상태로만 판단하는것도 가능할지 궁금해요!

Ivoryeee replied:
`activeMoribSet`로만 판단 시 빌드 에러가 발생해 `activeMoribSetId`를 추가해 뒀습니다!

suwonthugger commented on src/shared/constants/colorPallete.ts:
```diff
@@ -0,0 +1,14 @@
+export const colors = [
+	'#FF8080',
+	'#FFB62F',
+	'#FFF787',
+	'#B6FFA5',
+	'#5CE082',
+	'#179F62',
+	'#06FFD2',
+	'#27C5FF',
+	'#3D6DFF',
+	'#7742FF',
+	'#FF74F8',
+	'#868C93',
+];
```
p3: 이 컬러를 사용하는 원모양의 아이콘은 자주 재사용할것 같은데 상아님이 색상과 사이즈를 받아서 사용할수 있게 shared에 공통 컴포넌트로 만들어주실수 있을까요?!

KIMGEONHWI replied:
> p3: 이 컬러를 사용하는 원모양의 아이콘은 자주 재사용할것 같은데 상아님이 색상과 사이즈를 받아서 사용할수 있게 shared에 공통 컴포넌트로 만들어주실수 있을까요?!

p3: 타이머페이지에서도 사용될 예정이라 좋은 의견인 것 같습니다!

Ivoryeee replied:
좋은 의견이에요! CircleColorIcon이라는 공통 컴포넌트를 생성해 두었습니다! :)

suwonthugger commented on src/components/templates/MoribSetPageWrapper/index.tsx:
```diff
@@ -0,0 +1,11 @@
+import { ReactNode } from 'react';
+
+interface MoribSetPageWrapperProps {
+	children: ReactNode;
+}
+
+const MoribSetPageWrapper = ({ children }: MoribSetPageWrapperProps) => {
+	return <div className="relative flex h-screen w-screen bg-gray-bg-01">{children}</div>;
+};
+
+export default MoribSetPageWrapper;
```
p3: 이제 atomic 패턴을 사용하지 않으니 요건 컴포넌트로 분리하지 말고 바로 적용해주세요~~

Ivoryeee replied:
네 수정해 두었습니다!

suwonthugger commented on src/pages/HomePage/components/SideBarHome.tsx:
```diff
@@ -13,17 +17,25 @@ import { ModalContentsSetting } from './ModalContents';
 	const openSettings = () => {
 		modalRef.current?.open();
 	};
+	const navigateMoribSet = () => {
+		navigate(ROUTES_CONFIG.moribSet.path);
+	};
```
p5: 이렇게 prop에 들어갈 함수를 정의해서 사용하는 습관을 들이는게 명확하고 보기 좋은것 같아요. 하나 배워갑니다~

suwonthugger commented on src/pages/MoribSetPage/components/CategoryMoribSet.tsx:
```diff
@@ -0,0 +1,87 @@
+				<IconPlus className="cursor-pointer" onClick={addMoribSet} />
+			</div>
+
+			<div className="mx-[1.8rem] gap-[10px]">
```
p5: gap에만 px를 적용한 이유가 궁금해요~~

Ivoryeee replied:
rem으로 작성하는 게 맞아 이 부분 수정했습니다!

suwonthugger commented on src/pages/MoribSetPage/components/Input/InputMoribSetUrl.tsx:
```diff
@@ -0,0 +1,80 @@
+const InputMoribSetUrl = ({ addUrl }: InputMoribSetUrlProps) => {
+	const inputRef = useRef<HTMLInputElement | null>(null);
+	const [showClearBtn, setShowClearBtn] = useState(false);
+	const [alert, setAlert] = useState({ error: '' });
```
p3: alert라는 ui가 있어서 의미가 조금 혼동될수도있을것 같아요, alert라는 네이밍 대신 error ,errorMessage등의 네이밍이 어떤지 제안드리고싶어요.

KIMGEONHWI replied:
p3: 저도 동의합니다. alert창이 어디있나 찾았었네요.

Ivoryeee replied:
동의합니다! 요 부분 `errorMessage`로 수정해 두었습니다!

suwonthugger commented on src/pages/MoribSetPage/components/Input/InputMoribSetUrl.tsx:
```diff
@@ -0,0 +1,80 @@
+	const errorStyle = 'border-error-02 border-[1px]';
+
+	return (
+		<form action="" onSubmit={handleSubmit}>
```
p5: form action을 빈문자열로 둔 이유가 궁금해요

Ivoryeee replied:
action 속성이 필요치 않아 삭제해 두었습니다. 꼼꼼한 확인 감사드려요!

suwonthugger commented on src/pages/MoribSetPage/components/Box/BoxMakeMoribSet.tsx:
```diff
@@ -0,0 +1,102 @@
+		setIsPaletteOpen(false);
+	};
+
+	const handleMoribNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
p1: 저희 컨벤션에서 요런건 import해서 사용하기로했으니 React.이 아닌 ChangeEvent를 import해서 사용 부탁드려유

Ivoryeee replied:
네 수정해 두었습니다!

suwonthugger commented on src/pages/MoribSetPage/components/Box/BoxRecommendService.tsx:
```diff
@@ -0,0 +1,88 @@
+					style={{
+						background: 'linear-gradient(90deg, rgba(51, 55, 63, 0.00) 0%, #33373F 100%)',
+					}}
+				/> */}
```
p3: 이건 어떤 주석인지 궁금해요!

Ivoryeee replied:
앗 주석 수정한다는 걸 놓쳤었네요 div 태그 내부에 `bg-gradient-to-r from-transparent to-[#33373F]` 로 추가해 두었습니다!

suwonthugger replied:
p5: 오호 gradient 적용 팁 하나 배워갑니다~

suwonthugger commented on src/pages/MoribSetPage/components/CategoryMoribSet.tsx:
```diff
@@ -0,0 +1,87 @@
+									<div
+										className="h-[1.4rem] w-[1.4rem] rounded-[70px]"
+										style={{ backgroundColor: moribSet.selectedColor }}
+									/>
```
p1: 인라인 스타일 대신 tailwind로 스타일 지정 부탁드려요~~

Ivoryeee replied:
수정했습니다! 당시 색상별 네이밍이 되어 있지 않아 테일윈드 특성상 컬러코드를 동적 클래스로 불러오는데 어려움이 있어, 해당 부분을 문의해 두고 인라인 스타일로 지정해 두었습니다. 현재는 네이밍이 지정돼 tailwind tailwind.config.js에 추가 후 테일윈드로 스타일로 작성했습니다!

suwonthugger reviewed (commented):
고생 많으셨습니다~~ ! 전체적인 네이밍도 모립세트 대신 AllowedService 로 변경 부탁드릴게요!

KIMGEONHWI commented on src/pages/MoribSetPage/MoribSetPage.tsx:
```diff
@@ -0,0 +1,139 @@
+
+	return (
+		<MoribSetPageWrapper>
+			<SideBarHome />
```
p3: SideBarHome의 경우 대원님이 Layout으로 설정해두어서 온보딩뷰 pr 머지되면, 삭제해도 괜찮을거 같아요!

Ivoryeee replied:
삭제해 두었습니다!

KIMGEONHWI commented on src/pages/MoribSetPage/MoribSetPage.tsx:
```diff
@@ -0,0 +1,139 @@
+				<button>
+					<BellIcon className="rounded-[1.6rem] hover:bg-gray-bg-04 active:bg-gray-bg-05" />
+				</button>
+			</div>
```
p3: 현재 `FriendSettingIcon`과 `BellIcon`이 별도의 `div`태그에 감싸고 `absolute`주었는데, 
`BoxMakeMoribSet`와 `BoxRecommendService`를 감싸고 있는 `div`태그에 포함시키는 것은 어떨까요?
관련된 UI 요소들을 하나의 컨테이너 안에 포함되므로 레이아웃 관리도 수월할 것 같습니다.

Ivoryeee replied:
FriendSettingIcon과 BellIcon이 일종의 `navBar` 역할을 한다고 생각해 분리해 두었는데, 다른 분들은 어떻게 생각하시는지 궁금합니다! @suwonthugger @seueooo

KIMGEONHWI commented on src/pages/MoribSetPage/components/CategoryMoribSet.tsx:
p1: 현재 첨부한 사진과 같이 `CategoryMoribSet` 컴포넌트의 `height`값 이상으로 모립세트 생성시 `height`값을 오버해서 모립세트가 생성되고 있습니다. 해당 부분 침범하지 않고, 스크롤이 생길 수 있도록 수정해주세요!
<img width="230" alt="스크린샷 2024-12-12 오후 5 01 22" src="https://github.com/user-attachments/assets/41b7349a-da32-4d35-9884-f50c786fc94b" />

Ivoryeee replied:
네! `overflow-y-scroll` 속성을 적용해 해당 부분 수정해 두었습니다!

KIMGEONHWI commented on src/pages/MoribSetPage/components/CategoryMoribSet.tsx:
```diff
@@ -0,0 +1,87 @@
+						moribSet.urlList.length > maxIconsToShow ? moribSet.urlList.length - maxIconsToShow : 0;
+
+					return (
+						<div
```
p3: 해당 부분 별도의 컴포넌트로 분리하는 것이 어떨까요?? 의견이 궁금합니다.
<img width="251" alt="스크린샷 2024-12-12 오후 5 06 48" src="https://github.com/user-attachments/assets/1a536fa3-8a62-42ba-92ff-d2bb6c97f216" />

Ivoryeee replied:
`BoxMoribSetItem`으로 분리해뒀습니다!
빈 행을 삽입하거나 Dropdown을 기준으로 충분히 구조 파악이 된다고 생각했었었는데, 막상 코드리뷰 반영을 하다 보니 이 부분이 헷갈리더라고요. 별도의 컴포넌트로 분리할 때 오히려 각 컴포넌트의 역할을 명확히 하고 가독성이 개선된다 느껴 수정해 두었습니다.

KIMGEONHWI replied:
> `BoxMoribSetItem`으로 분리해뒀습니다! 빈 행을 삽입하거나 Dropdown을 기준으로 충분히 구조 파악이 된다고 생각했었었는데, 막상 코드리뷰 반영을 하다 보니 이 부분이 헷갈리더라고요. 별도의 컴포넌트로 분리할 때 오히려 각 컴포넌트의 역할을 명확히 하고 가독성이 개선된다 느껴 수정해 두었습니다.

확인했습니다. 명확한 근거 굿입니다.

KIMGEONHWI commented on src/pages/MoribSetPage/components/CategoryMoribSet.tsx:
```diff
@@ -0,0 +1,87 @@
+
+								<Dropdown.Root>
+									<Dropdown.Trigger>
+										<IconMeatBall className="cursor-pointer" />
```
p2: 현재 `IconMeatBall`클릭시, `onClick={() => setActiveMoribSetId(moribSet.id)}`도 함께 활성화되고 있는데 `event.stopPropagation()`적용해주어서 이벤트 전파 차단해주세요!

Ivoryeee replied:
네 수정해 두었습니다! 
IconMeatBall에 `e.stopPropagation()` 를 추가할 경우 `DropdownTrigger`의 `handleToggleOpen`이벤트까지 영향을 미치는 문제가 발생해 아래와 같이 `DropdownTrigger`에 `preventPropagation` `prop`을 추가하는 방향으로 수정했습니다.

```typescript
interface DropdownTriggerProps {
	children: ReactNode;
	preventPropagation?: boolean;
}

const DropdownTrigger = ({ children, preventPropagation }: DropdownTriggerProps) => {
	const { handleToggleOpen } = useDropdownContext();

	return (
		<button
			onClick={(e) => {
				if (preventPropagation) {
					e.stopPropagation();
				}
				handleToggleOpen();
			}}
			className="relative"
		>
			{children}
		</button>
	);
};
```

KIMGEONHWI commented on src/shared/constants/urlRegex.ts:
p1: 유효성 검사 로직 따로 생성안하고, 이미 존재하는 `/src/shared/utils/isUrlValid/index.ts`해당 파일 가져다 사용하는 것이 어떨까요?? 중복되는 것같네요.

Ivoryeee replied:
동의합니다! urlRegex의 유효성 규칙이 더 엄격하여 현재 파일은 삭제하고 isUrlValid의 정규식 부분분 수정하였습니다.

KIMGEONHWI reviewed (commented):
고생하셨습니다! 중복 유효성 검사도 추가해주면 좋을 것 같습니다.
<img width="914" alt="스크린샷 2024-12-12 오후 5 56 02" src="https://github.com/user-attachments/assets/a2026936-7294-452a-9d8c-a43747923ec4" />

seueooo reviewed (commented):
중복 유효성 검사만 추가해주시면 될 것 같아요! 저도 다른분들 코멘트에 동의합니다~! 고생하셨어여~~

KIMGEONHWI committed:
[ Style ] 모립세트 컴포넌트 구현 (#225)

* refactor: SideBoxTemporary 컴포넌트 삭제

* feat: MoribSetCOntainer 컴포넌트 구현

* style: 허용할 사이트 영역 홀,짝수번째 배경화면 다르게 수정

* feat: 취소 버튼 온클릭 핸들러 추가

* refactor: MoribSetContainer가 열린 상태에서 다른 요소 클릭 가능하게 수정

* feat: 배열 별도의 상수 파일로 분리 및 기능 구현

* refactor: 아이콘 url도 ALLOW_SITE_LIST에서 관리

* codereview: 대원 코드리뷰 반영

* refactor: 모립세트 워딩 허용서비스 세트로 수정

* codereview: 대원, 상아 코드리뷰 반영

* feat: 툴팁 생성

* refactor: 타이머페이지 툴팁 삭제

* refacotr: 툴팁 isAllowedServiceVisible이 true일 때만 화면에 등장하도록 수정

* codereview: BoxAllowedService에서 PopoverAllowedService로 네이밍 변경

seueooo committed:
[ Style ] 설정 뷰 내의 경고 모달 뷰 구현 (#227)

* feat: alert모달 내 버튼 공통 컴포넌트 생성 (#222)

* feat: ModalContentsAlert 생성 (#222)

* feat: input창 에러일 때 추가

* fix: 필요없는 코드 삭제 (#222)

* refactor: 약간의 로직분리 .. (#222)

* code review: user email prop 추가, 모달 컴포넌트 하나의 객체로 export 하도록 수정

* style: 스타일 누락된 것 추가

* refactor: userEmail props 추가 누락된 부분 수정

suwonthugger committed:
[ Style ] 온보딩 뷰 구현 (#228)

* refactor: home/ not found 페이지에 쓰이는 사이드바 컴포넌트 리팩토링 (#224)

- 세팅모달, 기존 SidebarHome에서 쓰이는 컴포넌트 구조 배럴파일 방식으로 변경
- HomePage, NotFoundePage 두 곳에서 쓰이므로 shared로 위치 이동
- Sidebar에서 쓰이는 svg는 Sidebar 폴더 아래로 이동

* refactor: 홈페이지 구조 absolute를 사용해서 변경 (#224)

* feat: 온보딩 페이지 라우팅 (#224)

* refactor: 설정 모달 위치 변경 (#224)

* refactor: 사이드바 위치 변경 (#224)

* refactor: 레이아웃 생성 및 적용 (#224)

* refactor: 폰트 적용 방식 자동완성 될 수 있도록 변경 (#224)

* feat: 사용하는 아이콘 다운로드 및 export 설정 (#224)

* feat: StartStep, FieldStep 구현 (#224)

* feat: ServiceStep 구현 및 export (#224)

* chore: 상수 분리 및 컴포넌트 디자인 일부 수정 (#224)

* feat: useFunnel 훅 생성 및 온보딩 페이지에 적용 (#224)

* feat: Step 컴포넌트 컨벤션에 맞게 네이밍 변경 (#224)

* feat: 온보딩 페이지 퍼블리싱 (#224)

* feat: input 상태 ServiceStep 컴포넌트에 연결 (#224)

* style: StepField 컴포넌트 화면 너비에 따라 패딩값 일부 수정 (#224)

* feat: ServiceStep 버튼에 disabled 속성 추가 및 엔터 눌렀을 때 입력되게 끔 로직 추가 (#224)

* feat: svg 아이콘 폴더 shared로 변경, assets 아래 페이지별 폴더를 만들어 관리 (#224)

* feat: 바뀐 svg들의 위치에 맞게 import 경로 변경 및 BoxAllowedService 버튼에 홈으로 이동하는 로직 추가 (#224)

* style: 반응형 고려하여 홈 페이지 컴포넌트 일부 스타일 수정 (#224)

- 미완성입니다요

* refactor: truncate 활용해서 스타일 재적용 (#224)

* refactor: StepField.tsx p태그에서 h2태그로 변경 (#224)

* chore: 브라우저 리스트 업데이트하면서 일부 패키지 버전업 (#224)

* refactor: StepService.tsx 인풋 상태 input -> inputURL로 수정 (#224)

suwonthugger committed:
[Refactor] home 페이지 폴더구조 변경 (#231)

* feat: home 페이지 폴더구조 변경 (#230)

* feat: 공통적으로 쓰이는 BoxTodo, ButtonTogoToggle -> shared로 이동 (#230)

* feat: shared로 이동한 컴포넌트 경로 변경 (#230)

* refactor: ModalContents 폴더 구조 변경 (#230)

* feat: import 문 구분을 위해 폴더 구조 배럴파일 구조로 변경 (#230)

* feat: commit 누락된 부분 저장 (#230)

* feat: 홈페이지 import 문 수정 (#230)

* refactor: atoms 컴포넌트 용도에 맞게 폴더 이동 및 네이밍 컨벤션에 맞게 변경 (#230)

* refactor: 바뀐 컴포넌트 경로에 맞게 import문 수정 (#230)

* refactor: 홈에서만 쓰이는 hook 폴더 이동 및 훅을 사용하는 컴포넌트의 import문 변경 (#230)

* refactor: ButtonSVG 사용 불필요하여 삭제 (#230)

* refactor: 상아 피드백 반영해서 UI 분류 페이지 제거 모든 컴포넌트 depth 1로 고정 (#230)

* refactor: hook import 경로 수정 (#230)

suwonthugger committed:
refactor: 로그인 페이지 구조 변경 및 불필요한 templates 레거시 코드 삭제 (#230) (#232)

suwonthugger committed:
[Refactor] 타이머 페이지 폴더 구조 변경 (#233)

* refactor: timer 컴포넌트/훅 컨벤션에 맞게 정리 및 불필요 컴포넌트/훅 삭제 (#230)

* refactor: 불필요한 templates 컴포넌트 삭제 (#230)

* Merge branch 'develop' of github.com:morib-in/Morib-Client into refactor/#230/timer-folder-structure

suwonthugger committed:
[Refactor] 온보딩 페이지 폴더구조 변경 (#234)

* refactor: 온보딩 페이지 컴포넌트 폴더 삭제 및 컴포넌트명으로 폴더 생성하여 적용 (#230)

* refactor: 상아 피드백 반영해서 컴포넌트 depth 1로 고정 (#230)

Ivoryeee committed:
[style] 컬러팔레트 네이밍 추가 및 색상 공통 컴포넌트 생성 (#220)

Ivoryeee committed:
[refactor] 색상 네이밍에 맞춰 컬러 팔레트 수정 (#220)

Ivoryeee committed:
[feat] url 중복 유효성 검사 로직 추가 (#220)

Ivoryeee committed:
[refactor] 드롭다운 클릭 시 이벤트 전파 방지 (#220)

Ivoryeee committed:
[style] 카테고리 내 허용서비스 컴포넌트 분리 (#220)

Ivoryeee committed:
[code review] 코드리뷰 반영 (#220)

Ivoryeee committed:
[style] moribSet -> allowedService 네이밍 변경 (#220)

Ivoryeee committed:
[chore] 리베이스 충돌 해결 (#220)

Ivoryeee committed:
[style] sideBar 삭제 후 Layout 적용 (#220)

suwonthugger commented on src/pages/MoribSetPage/components/CategoryMoribSet.tsx:
```diff
@@ -54,19 +51,22 @@ const CategoryMoribSet = ({
-									<Dropdown.Trigger>
-										<IconMeatBall className="cursor-pointer" />
+									<Dropdown.Trigger preventPropagation={true}>
+										<IconMeatBall className="cursor-pointer hover:rounded-full hover:bg-gray-bg-05" />
```
p5: 단순 궁금한 부분인데, 어떤 부분을 클릭했을 때 이벤트 propagation이 일어났나요?!

Ivoryeee replied:
상위 div영역 즉 카드를 클릭할 때 focus가 됩니다! dropdown이 카드 내부에 위치해 있어 삭제 버튼을 누를 때 해당 카드 영역까지 포커스 되는 문제가 있어 dropdown 영역 클릭 시 이벤트가 캡처링되지 않도록 수정했습니다

suwonthugger commented on src/pages/MoribSetPage/components/Input/InputAllowedServiceUrl.tsx:
```diff
@@ -1,6 +1,6 @@
 import { FormEvent, useRef, useState } from 'react';
 
-import ButtonSVG from '@/shared/components/ButtonSVG';
+import ButtonSVG from '@/shared/components/Button/ButtonSVG/ButtonSVG';
```
p1: ButtonSVG는 불필요하여 삭제하기로했어요~ ButtonSVG 대신 button으로 변경해주시고, 기존에 사용하던 ButtonSVG 컴포넌트는 삭제 부탁드립니당

Ivoryeee replied:
오호 확인했습니다! 반영해 두겠습니다!

Ivoryeee replied:
불필요한 컴포넌트라 삭제하셨던 거군요! 넵 이 부분 반영해 두었습니다!

suwonthugger commented on src/pages/MoribSetPage/components/Box/BoxAllowedServiceItem.tsx:
```diff
@@ -0,0 +1,71 @@
+	return (
+		<div
+			key={allowedService.id}
+			className={`mb-[8px] h-[8rem] w-[36.6rem] flex-col items-start justify-end rounded-[8px] bg-gray-bg-01 p-[1.4rem] ${activeAllowedServiceId === allowedService.id ? 'border-[1px] border-mint-01' : ''}`}
```
p1: 여기 mb에 rem으로 변경해주세요~~

Ivoryeee replied:
수정해두었습니다!

Ivoryeee replied:
수정해두었습니다!

suwonthugger commented on src/pages/MoribSetPage/AllowedServicePage.tsx:
```diff
@@ -0,0 +1,135 @@
+				/>
+
+				<div className="ml-[4.2rem] mt-[6.8rem]">
+					{activeAllowedServiceId && activeAllowedService && (
+						<BoxMakeAllowedService
+							allowedService={activeAllowedService}
+							updateAllowedService={(key, value) => handleUpdateAllowedService(activeAllowedServiceId, key, value)}
+						/>
+					)}
```
p2: 생각해보니 BoxMakeAllowedService 라는 컴포넌트를 보여주지 않는 경우가 없을것 같다고 생각이 드는데, 혹시나 제 생각이 맞다면 조건문 지워도 괜찮을것 같아요

suwonthugger replied:
p5: 그리고 궁금한게 activeAllowedServiceId 가 null이 되는 경우는 어떤 경우인가요?!

Ivoryeee replied:
반영해 두었습니다 
항상 하나씩 띄워지게 되는 allowedService는 실제 값이 아니라 undefine이나 null으로 설정되는 경우가 있다고 생각했습니다 activeAllowedService는 기본 값이외의 입력이 필요한데 activeAllowedServiceId는 항상 특정 카드를 포커스하고 있어 number로만 판단이 가능하겠다 싶어 이 부분 수정했습니다!

suwonthugger commented on src/pages/MoribSetPage/AllowedServicePage.tsx:
```diff
@@ -0,0 +1,135 @@
+
+	return (
+		<div className="flex h-screen w-screen bg-gray-bg-01">
+			<div className={`absolute right-[4.4rem] top-[5.4rem] flex gap-[0.8rem]`}>
+				<button onClick={handleOpenFriendsModal}>
+					<FriendSettingIcon className="rounded-[1.6rem] hover:bg-gray-bg-04 active:bg-gray-bg-05" />
+				</button>
+				<button>
+					<BellIcon className="rounded-[1.6rem] hover:bg-gray-bg-04 active:bg-gray-bg-05" />
+				</button>
+			</div>
```
p1: 친구, 벨 아이콘 부분에서 right 값은 제대로 준것 같은데 제 웹에서는 아래와 같이 잘려보이네요...!  relative가 적용된 상위 태그 기준으로 absolute right값이 어떻게 적용되었는지 한번 확인 부탁드립니다~
<img width="512" alt="image" src="https://github.com/user-attachments/assets/c748d525-0c7a-42d6-ba3d-a9a717a1c140" />

Ivoryeee replied:
rebase 후 Layout을 적용하면서 이 부분을 놓친 것 같네요.. 요 부분분 수정해 두었습니다!

suwonthugger commented on src/pages/MoribSetPage/AllowedServicePage.tsx:
```diff
@@ -0,0 +1,135 @@
+		);
+	};
+
+	useEffect(() => {
+		if (allowedServices.length === 0) {
+			setActiveAllowedServiceId(null);
+		} else if (!allowedServices.find((set) => set.id === activeAllowedServiceId)) {
+			setActiveAllowedServiceId(allowedServices[allowedServices.length - 1].id);
+		}
+	}, [allowedServices, activeAllowedServiceId]);
```
p5: 이 부분 확인 중에 있는데 상아님 말씀대로 useEffect 없이 하위 컴포넌트에서 변경된 상태를 제대로 받지 못하네요... 원인이 뭔지 파악해보겠습니다.

suwonthugger reviewed (commented):
고생 많으셨습니다!

Ivoryeee committed:
[code review] 코드리뷰 반영 (#220)

suwonthugger commented on src/shared/layout/components/Sidebar/Sidebar.tsx:
```diff
@@ -0,0 +1,39 @@
+	return (
+		<>
+			<aside className="sticky top-0 flex min-h-screen w-[7.4rem] flex-shrink-0 flex-col items-center justify-between bg-gray-bg-02 pb-[2.1rem] pt-[5.4rem]">
+				<button type="button" className="relative flex w-full items-center gap-[0.8rem]">
+					<hr className="h-[54px] w-[2px] rounded-r-[8px] bg-white" />
+					<LogoIcon />
+				</button>
```
p1: 리베이스 되면서 누락되었는데 여기에 허용서비스뷰 라우팅 부탁드립니당!

suwonthugger commented on src/shared/components/Dropdown.tsx:
```diff
@@ -50,13 +50,22 @@ const DropdownRoot = ({ children }: DropdownRootProps) => {
 
-const DropdownTrigger = ({ children }: DropdownTriggerProps) => {
+const DropdownTrigger = ({ children, preventPropagation }: DropdownTriggerProps) => {
 	const { handleToggleOpen } = useDropdownContext();
 
 	return (
-		<button onClick={handleToggleOpen} className="relative">
+		<button
+			onClick={(e) => {
+				if (preventPropagation) {
+					e.stopPropagation();
+				}
+				handleToggleOpen();
+			}}
+			className="relative"
+		>
```
p3: `e.stopPropagation();`은 조건부로 받아서 적용안하고, 이벤트 전파를 방지하기 위해서 항상 적용해주어도 괜찮을것 같아요!

suwonthugger reviewed (commented):
반영 확인했습니다~ 저는 마지막으로 아래 사항만 반영해주시면 괜찮을것 같아요

Ivoryeee committed:
[style] 사이드바 홈, 허용서비스뷰 추가 (#220)

Ivoryeee committed:
[style] 드롭다운 이벤트 전파 방지 적용 (#220)

suwonthugger committed:
Merge branch 'develop' into style/#220/moribSet-view

suwonthugger reviewed (approved):
LGTM

seueooo reviewed (approved):
LGTM~!!
